app/assets/javascripts/oregon_digital/add_to_my_shelf.js

```diff
@@ -0,0 +1,5 @@
+$( document ).ready(function() {
+  $('#Select_all_on_page').click(function() {
+    $('.my_shelf_result_checkbox').each(function () { this.checked = !this.checked; });
```

Comment:
This doesn't seem to work as expected for a mix of already checked and unchecked boxes

Reply:
Yeah so if some are checked, and some arent, we check all the unchecked ones and leave the others. If all are checked, then uncheck them. That sound about right?

Reply:
Sounds right, but complicated. I think we get the same outcome by setting all .my_shelf_result_checkbox checked values to #Select_all_on_page checked value

Reply:
Yeah I see what you are saying. we should be able to push the #Select_all_on_page object through then check its state. 👍